docs/how-to/contribute-a-builtin.md

```diff
@@ -0,0 +1,68 @@
+# How to contribute a Builtin plugin
+
+Within a [RuleSet YAML file](../../README.md#ruleset), scriptlet probes are an escape hatch for complex, programmatic assertions. They are not intended to assert against common model patterns. For example, scriptlet probe authors may assert that certain applications or relations exist. This repetitiveness can be abstracted into Builtin assertions. Refer to this sample [RuleSet with Builtin assertions](../../tests/resources/probes/ruleset/builtins.yaml) as a starting point.
+
+Builtins are probes, serving as complementary (or isolated) assertions, depending on the scope of their assertions.
+
+```yaml
+name: RuleSet - demo
+probes:
+  - name: Probe - test passing
+    type: scriptlet
+    url: file://tests/resources/probes/python/passing.py
+  - name: My builtin
+    type: builtin/my-builtin
+    with:
+      - foo: one
+      - foo: two
+```
+
+## Builtin plugins
+
+Juju-doctor is designed to accept builtin plugins via contributions made in the [src.juju_doctor.builtin](../../src/juju_doctor/builtin/) directory. Each of these files are a builtin and the name of the plugin is derived from the Python file name. Additionally, each builtin conforms to the [scriptlet interface](../../README.md#scriptlet):
+
+```python
+# src/juju_doctor/builtins/my-builtin.py
+class FooModel(BaseModel):
+    foo: str
+
+def status(juju_statuses: Dict[str, Dict], **kwargs):
+    assert kwargs["with_args"], "No arguments were provided"
+    foo_models = [FooModel(**foo) for foo in kwargs["with_args"]]
+    for foo_model in foo_models:
+        # Run your assertion on each foo_model (FooModel) here ...
+```
+
+This would create the following API in a RuleSet:
```

Comment:
Not sure iiuc, but the builtin does doesn't create anything, it only enables one to use the following API in a ruleset..?